storm/src/main/storm/mesos/NimbusMesosScheduler.java

```diff
@@ -109,6 +119,28 @@ public void statusUpdate(SchedulerDriver driver, TaskStatus status) {
     }
   }
 
+  private void updateLogviewerState(TaskStatus status) {
+    String nodeId = status.getTaskId().getValue().split("\\" + DEFAULT_MESOS_COMPONENT_NAME_DELIMITER)[0];
+    switch (status.getState()) {
+      case TASK_STAGING: return;
+      case TASK_STARTING: return;
+      case TASK_RUNNING: return;
+      default:
+    }
+    // if it gets to this point it means logviewer terminated; update ZK with new logviewer state
+    String logviewerZKPath = String.format("%s/%s", logviewerZkDir, nodeId);
+    if (zkClient.nodeExists(logviewerZKPath)) {
+      LOG.info("updateLogviewerState: Remove logviewer state in zk: {}", logviewerZKPath);
+      zkClient.deleteNode(logviewerZKPath);
+      LOG.info("updateLogviewerState: Reviving offers for logviewer relaunch");
+      mesosNimbus._driver.reviveOffers();
+      StormSchedulerImpl stormScheduler = (StormSchedulerImpl) mesosNimbus.getForcedScheduler();
+      stormScheduler.unsetOffersSuppressed();
+    } else {
+      LOG.error("Task exists for logviewer that isn't tracked in ZooKeeper");
```

Comment:
Also, can you please print out the Task ID whenever you are emitting logs about a particular task?  I realize it might feel redundant with the other log that got printed right before this in the `statusUpdate()` method, however it's helpful to be able to individually grep lines in log files.  I really wish the implementors of storm-kafka had added more information into each log line, especially because many lines get interleaved and come from different spout threads running in the same worker process, so it's impossible to figure out which partition and/or topic is being referred to in various storm-kafka log lines.